hd-wallet/pluginv2.lua

```diff
@@ -0,0 +1,511 @@
+        hex_key = offset..hex_key
+    end
+    return hex_key
+end
+-- return scalar addition of point
+function add_public_keys(k1, k2)
+    local secP256K1 = EcGroup.from_name('SecP256K1')
+  	local pt_1 = secP256K1:point_from_binary(Blob.from_hex(k1))
+  	local pt_2 = secP256K1:point_from_binary(Blob.from_hex(k2))
+  	local p1 = secP256K1:point_from_components(pt_1:x(), pt_1:y())
+    local p2 = secP256K1:point_from_components(pt_2:x(), pt_2:y())
+    local p3 = p1 + p2
+    return compress_public_key(p3:x():to_bytes_be():hex(), p3:y():to_bytes_be():hex())
+end
+-- checksum: double sha256 of serialized key
+function get_check_sum(child_key)
+    local child_key_string = child_key.version.. child_key.depth..  child_key.fingerprint.. child_key.index.. child_key.chain_code.. child_key.key
+    local sha256_hash1 = assert(digest { data = Blob.from_hex(child_key_string), alg = 'SHA256' }).digest:hex()
+    local sha256_hash2 = assert(digest { data =  Blob.from_hex(sha256_hash1), alg = 'SHA256' }).digest:hex()
+    return sha256_hash2
+end
+-- derive new child key from parent key
+function derive_new_child(parent_key, child_idx)
+    local data = ""
+    -- if index is greater than equal to first hardend key
+    if tonumber(child_idx) >= FIRST_HARDENED_CHILD then
+        data = parent_key.key
+    else
+        -- parent is private
+        -- data equal to public key of parent private
+        if parent_key.version == PRIVATE_WALLET_VERSION then
+            data = public_key_for_private_key(string.sub(parent_key.key, 3, 66))
+        else
+            -- key is public
+            -- data equal to parent key
+           data = parent_key.key
+        end
+    end
+    
+    -- concatenate index into data
+    local index_hex = num_2_hex(child_idx, 8)
+    data = data..index_hex
+    hmac = get_hmac(parent_key.chain_code, data)
+
+    child_key = {
+        index = index_hex,
+        chain_code = string.sub(hmac, 65, 128),
+        depth = num_2_hex(tonumber(parent_key.depth + 1), 2),
+        is_private = parent_key.is_private,
+    }
+  
+    if parent_key.version == PRIVATE_WALLET_VERSION then
+        child_key.version = PRIVATE_WALLET_VERSION 
+        fingerprint = hash160(public_key_for_private_key(string.sub(parent_key.key, 3, 66))) 
+        child_key.fingerprint = string.sub(fingerprint, 1, 8)
+        -- appending 00 to make key size 33 bit
+        child_key.key = "00"..tostring(add_private_keys(string.sub(hmac, 1, 64), parent_key.key))
+    else
+        child_key.version = PUBLIC_WALLET_VERSION
+        fingerprint = hash160(parent_key.key)
+        child_key.fingerprint = string.sub(fingerprint, 1, 8)
+        key_bytes = public_key_for_private_key(string.sub(hmac, 1, 64))
+        child_key.key = add_public_keys(key_bytes, parent_key.key)
+    end
+    
+   child_key.checksum = string.sub(get_check_sum(child_key), 1, 8)
+  
+    return child_key
+end
+-- serialize child key and encode into base58
+function serialize(child_key)
```

Comment:
Now this is only used once, and to compute a value that is unused (`child_key_serialized`). It can be removed.

Reply:
Done, updated